packages/web-components/src/calendar/calendar.ts

```diff
@@ -0,0 +1,368 @@
+      ? 'slotted-link inactive'
+      : 'slotted-link';
+  }
+
+  /**
+   * Creates a class string for cells on the right panel
+   * @returns - string of class names
+   * @public
+   */
+  public getRightCellClassNames(detail: number, todayMonth: number, todayYear: number) {
+    const isToday = this.yearPickerOpen ? detail === todayYear : detail === todayMonth;
+
+    return ['right-cell-outer', this.hasAttribute('highlightCurrentMonth') && isToday && 'right-panel-today']
+      .filter(Boolean)
+      .join(' ');
+  }
+
+  /**
+   * Returns a list of month labels
+   * @returns A 2D array of month texts
+   * @public
+   */
+  public getMonthText(): { text: string; detail: number }[][] {
+    const months = this.dateFormatter.getMonths();
+    const monthsText: { text: string; detail: number }[][] = [];
+    let monthCount = 0;
+
+    while (monthCount < months.length || monthsText[monthsText.length - 1].length % 4 !== 0) {
+      const month = { text: months[monthCount], detail: monthCount + 1 };
+      const target = monthsText[monthsText.length - 1];
+      if (monthsText.length === 0 || target.length % 4 === 0) {
+        monthsText.push([month]);
+      } else {
+        target.push(month);
+      }
+      monthCount++;
+    }
+    return monthsText;
+  }
+
+  /**
+   * Returns a list of year labels for a decade
+   * @returns A 2D array of year texts
+   * @public
+   */
+  public getDecadeText(decadeStartYear: number): { text: string; detail: number }[][] {
+    const decade = this.dateFormatter.getDecade(decadeStartYear);
+    const decadeText: { text: string; detail: number }[][] = [];
+    let yearCount = 0;
+
+    while (yearCount < decade.length || decadeText[decadeText.length - 1].length % 4 !== 0) {
+      const yearText = { text: decade[yearCount], detail: decadeStartYear + yearCount };
+      const target = decadeText[decadeText.length - 1];
+      if (decadeText.length === 0 || target.length % 4 === 0) {
+        decadeText.push([yearText]);
+      } else {
+        target.push(yearText);
+      }
+      yearCount++;
+    }
+    return decadeText;
+  }
+
+  /**
+   * Updates calendar to show today when user clicks on "Go to today"
+   * @param event - mouse event for clicking on the link
+   */
+  public handleGoToToday(event: MouseEvent) {
+    const today: Date = new Date();
+    this.switchMonth(today.getMonth() + 1, today.getFullYear());
+    this.yearPickerOpen = false;
+  }
+
+  /**
+   * Emits the "rightcellselected" event with the seleced month.
+   * @param cellDetail - The month of year that's been selected
+   * @public
+   */
+  public handleRightCellSelect(event: Event, cellDetail: number): void {
+    event.preventDefault;
+    this.$emit('rightcellselected', cellDetail);
+  }
+
+  /**
+   * Handles selecting dates on the calendar's date view
+   * Stores the selected dates in the selected-dates attribute
+   * @param event - 'dateselected' event
+   */
+  public dateSelectedHandler(event: any) {
```

Comment:
With this approach, I run into the same problem of not being able to add an event listener in the connectedCallBack, because the addEventListener method can't take custom events. I asked Chris for his advice but we decided that we can leave it as type "any" if it works. But I definitely understand the type safety concern. Would love to get your thoughts on this!

Reply:
Sounds good! I'm not beholden to my comments above. If Chris is cool leaving it as "any" I am too.